test/models/drive_test.dart

```diff
@@ -0,0 +1,489 @@
+    });
+  });
+  group('Drive.getMaxUsedSeats', () {
+    test('can handle null', () async {
+      final Drive drive = DriveFactory().generateFake(createDependencies: false);
+      expect(drive.getMaxUsedSeats(), null);
+    });
+    test('no seats taken', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByDriver,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 0);
+    });
+    test('seats are taken by approved rides', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByRider,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.rejected,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 15, 10),
+            endTime: DateTime(2022, 2, 2, 15, 20),
+            status: RideStatus.approved,
+            seats: 1),
+      ]);
+      expect(drive.getMaxUsedSeats(), 2);
+    });
+    test('multiple rides take seats at the same time', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14),
+            endTime: DateTime(2022, 2, 2, 14, 20),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 20),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 3);
+    });
+  });
+  group('Drive.isRidePossible', () {
+    test('can handle empty drive', () async {
```

Comment:
Please remove `can handle` from these descriptions. You could write these before any test case and I don't think it adds any value (and in this case, it's inconsistent with the description right below)

Reply:
Mh okay I suspect we'll have to unify that anyway some time